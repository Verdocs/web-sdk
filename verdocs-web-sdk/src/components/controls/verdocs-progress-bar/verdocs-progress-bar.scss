@import '../../../theme.scss';

verdocs-progress-bar {
  font-family: $primary-font;
  width: 100%;
  display: flex;
  flex-wrap: nowrap;
  box-sizing: border-box;
  flex-direction: column;

  .labels {
    display: flex;
    margin: 0 0 8px 0;
    flex-direction: row;
    justify-content: space-between;

    .label {
      display: flex;
      font-size: 14px;
      font-weight: 600;
      margin-right: 8px;
      color: $verdocs-grey-1;
    }
  }

  .bar {
    display: flex;
    flex: 0 0 10px;
    border-radius: 5px;
    background-color: $verdocs-bg-1;

    .slider {
      display: flex;
      border-radius: 5px;
      background-color: $verdocs-green;
    }
  }
}
