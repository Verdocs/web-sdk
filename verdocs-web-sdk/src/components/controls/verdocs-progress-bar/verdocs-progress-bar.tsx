import {Component, Prop, h, Host} from '@stencil/core';

/**
 * A simple progress bar.
 */
@Component({
  tag: 'verdocs-progress-bar',
  styleUrl: 'verdocs-progress-bar.scss',
  shadow: false,
})
export class VerdocsProgressBar {
  /**
   * Optional label to display above the bar
   */
  @Prop() label: string = '';

  /**
   * If true, the progress percentage will be displayed above the bar.
   */
  @Prop() showPercent: boolean = false;

  /**
   * The current progress value (0-100)
   */
  @Prop() percent: number = 0;

  render() {
    const widthPercent = Math.ceil((100 * Math.min(this.percent, 100)) / 100);
    console.log('Rendering progress bar', {widthPercent, label: this.label, showPercent: this.showPercent, percent: this.percent});

    return (
      <Host>
        <div class="labels">
          {this.label && <div class="label">{this.label}</div>}
          {this.showPercent && <div class="label">{this.percent}%</div>}
        </div>

        <div class="bar">
          <div class="slider" style={{width: `${widthPercent}%`}}></div>
        </div>
      </Host>
    );
  }
}
