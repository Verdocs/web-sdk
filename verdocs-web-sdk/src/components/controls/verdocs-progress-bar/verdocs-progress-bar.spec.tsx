import { newSpecPage } from '@stencil/core/testing';
import { VerdocsProgressBar } from './verdocs-progress-bar';

describe('verdocs-progress-bar', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsProgressBar],
      html: `<verdocs-progress-bar></verdocs-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with label and percent', async () => {
    const page = await newSpecPage({
      components: [VerdocsProgressBar],
      html: `<verdocs-progress-bar label="Uploading..." percent="54"></verdocs-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.rootInstance.label).toBe('Uploading...');
    expect(page.rootInstance.percent).toBe(54);
  });

  it('renders with showPercent true', async () => {
    const page = await newSpecPage({
      components: [VerdocsProgressBar],
      html: `<verdocs-progress-bar percent="75"></verdocs-progress-bar>`,
    });
    (page.rootInstance as any).showPercent = true;
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    expect(page.rootInstance.showPercent).toBe(true);
    expect(page.rootInstance.percent).toBe(75);
  });
});
