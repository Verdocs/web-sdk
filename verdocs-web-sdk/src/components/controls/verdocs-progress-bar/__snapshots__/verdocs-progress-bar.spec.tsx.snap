// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-progress-bar renders with default props 1`] = `
<verdocs-progress-bar>
  <div class="labels"></div>
  <div class="bar">
    <div class="slider" style="width: 0%;"></div>
  </div>
</verdocs-progress-bar>
`;

exports[`verdocs-progress-bar renders with label and percent 1`] = `
<verdocs-progress-bar>
  <div class="labels">
    <div class="label">
      Uploading...
    </div>
  </div>
  <div class="bar">
    <div class="slider" style="width: 54%;"></div>
  </div>
</verdocs-progress-bar>
`;

exports[`verdocs-progress-bar renders with showPercent true 1`] = `
<verdocs-progress-bar>
  <div class="labels">
    <div class="label">
      Loading...
    </div>
    <div class="label">
      75%
    </div>
  </div>
  <div class="bar">
    <div class="slider" style="width: 75%;"></div>
  </div>
</verdocs-progress-bar>
`;
